tools/nebula-flink/src/main/java/org.apache.flink/connector/nebula/catalog/AbstractNebulaCatalog.java

```diff
@@ -6,20 +6,248 @@
 
 package org.apache.flink.connector.nebula.catalog;
 
+import com.facebook.thrift.TException;
+import com.google.common.net.HostAndPort;
+import com.vesoft.nebula.client.meta.MetaClientImpl;
+import org.apache.flink.connector.nebula.table.NebulaDynamicTableFactory;
+import org.apache.flink.connector.nebula.utils.NebulaUtils;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.catalog.*;
+import org.apache.flink.table.catalog.exceptions.*;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatistics;
+import org.apache.flink.table.catalog.stats.CatalogTableStatistics;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.factories.Factory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
 /**
  * AbstractNebulaCatalog is used to get nebula schema
+ *
+ * flink-nebula catalog doesn't support write operators， such as create/alter
  */
-public interface AbstractNebulaCatalog {
+public abstract class AbstractNebulaCatalog extends AbstractCatalog {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractNebulaCatalog.class);
+
+    protected final String username;
+    protected final String password;
+    protected final String address;
+
+
+    public AbstractNebulaCatalog(String catalogName, String defaultDatabase, String username, String pwd, String address) {
+        super(catalogName, defaultDatabase);
+        // TODO check arguements
+        this.username = username;
+        this.password = pwd;
+        this.address = address;
+    }
+
+
+
+    @Override
+    public void open() throws CatalogException{
+        // test graphClient connection
```

Comment:
test  meta client ? why?

Reply:
fail early if we cannot connect to nebula

Reply:
Actually  you create a meta client ...

Reply:
modify the comment to "test metaClient connection"